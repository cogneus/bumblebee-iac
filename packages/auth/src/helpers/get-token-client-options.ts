import { ClientOptions } from 'bumblebee-common'
import { env } from 'process'

export const getTokenClientOptions = (): ClientOptions => {
  const {
    RESOURCE_NAME_PREFIX,
    KMS_ID,
    SECRET_ARN_PREFIX,
    SECRET_SIGNING_KEY_NAME,
    SECRET_TOKENS_NAME,
    SECRET_USERS_NAME,
  } = env
  return {
    secrets: {
      idPrefix: SECRET_ARN_PREFIX,
      prefix: RESOURCE_NAME_PREFIX,
      names: {
        tokens: { name: SECRET_TOKENS_NAME },
      },
      ids: {
        authSigningKey: { name: SECRET_SIGNING_KEY_NAME },
        users: { name: SECRET_USERS_NAME },
      },
      kmsId: KMS_ID,
    },
  }
}
