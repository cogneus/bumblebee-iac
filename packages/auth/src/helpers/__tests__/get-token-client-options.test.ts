import { getTokenClientOptions } from '../get-token-client-options'

describe('Get auth client options tests', () => {
  beforeEach(() => {
    jest.clearAllMocks()
    process.env = {
      AWS_REGION: 'eu-west-1',
      LOG_LEVEL: 'debug',
      NODE_ENV: 'dev',
      RESOURCE_NAME_PREFIX: 'test-prefix',
      SECRET_SIGNING_KEY_NAME: 'signingKey',
      TOKEN_AUDIENCE: 'audience',
      TOKEN_EXPIRY: 'expiry',
      TOKEN_ISSUER: 'issuer',
      SECRET_TOKENS_NAME: 'tokens',
      SSM_NAME_PREFIX: 'ssm-prefix',
      SSM_USERS_NAME: 'users',
    } as any
  })

  test('should return the value of the secrets key', async () => {
    const value = getTokenClientOptions()
    expect(value).toMatchSnapshot()
  })
})
