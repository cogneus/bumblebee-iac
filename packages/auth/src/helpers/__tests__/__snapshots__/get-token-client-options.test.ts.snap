// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Get auth client options tests should return the value of the secrets key 1`] = `
{
  "secrets": {
    "idPrefix": undefined,
    "ids": {
      "authSigningKey": {
        "name": "signingKey",
      },
      "users": {
        "name": undefined,
      },
    },
    "kmsId": undefined,
    "names": {
      "tokens": {
        "name": "tokens",
      },
    },
    "prefix": "test-prefix",
  },
}
`;
